aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs

```diff
@@ -161,22 +161,55 @@ public void OnToggleSettings()
     {
         if (_preventSettingsToggle) return;
 
-        if (_currentSceneName == _sceneNameStartingMenu) // Toggle from menu to menu settings and back.
+        IsOnSettings = !IsOnSettings;
+
+        switch (_currentSceneName)
         {
-            IsOnMenuSettings = !IsOnMenuSettings;
-            SettingMenuCanvas.SetActive(IsOnMenuSettings);
-            MenuCanvas.SetActive(!IsOnMenuSettings);
+            case _sceneNameStartingMenu:
+                MenuCanvas.SetActive(!IsOnSettings);
```

Comment:
Does this do what it is supposed to do? It is double negating, since you already flipped the bool before the switch case

Reply:
On startup the bool IsOnSettings is on false. When the settings button is pressed the bool is true and sets all things that need to be on on. However, when going into the settings menu needs to be off.

tl;dr It does what it needs to do